server/src/main/java/org/opensearch/action/ActionRequest.java

```diff
@@ -67,6 +70,14 @@ public boolean getShouldStoreResult() {
         return false;
     }
 
+    public void runAs(Subject runAs) {
```

Comment:
@reta @peternied What do you think of this latest commit which allows a caller to attach a subject to an ActionRequest? Look at the change below in AbstractClient as well which will surround the call to doExecute with the subject if its been explicitly set. 

You can find an example usage here: https://github.com/cwperks/security/blob/security-subject/src/integrationTest/java/org/opensearch/security/plugin/TransportIndexDocumentIntoSystemIndexAction.java#L56-L95

In that test it creates a system index using the plugin subject and tries to index a document into the system index using the user subject (which fails).

It shows how its possible to run transport actions with different identities.

Reply:
-1 to be fair, I think the `runAs` is only relevant in context of identity API, we should not expose it anywhere else.

Reply:
Would there be a way to potentially make this more generic? i.e. HTTPRequest has a notion of Headers and the AUTHORIZATION header is a special http header which carries auth info.

Reply:
Reverted this change and the one in AbstractClient